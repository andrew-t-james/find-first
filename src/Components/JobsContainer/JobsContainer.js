import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SlideMenu from 'react-slide-menu';
import { getSavedJobsFromFirebase } from '../../thunks/firebase';
import { nav } from '../../helpers/nav';
import { toggleMenu } from '../../Actions/menu';

import Header from '../Header/Header';
import SavedJobs from '../SavedJobs/SavedJobs';
import Loader from '../../Components/Loader/Loader';
import Footer from '../Footer/Footer';

export class JobsContainer extends Component {
  constructor() {
    super();
    this.state = {
      savedJobs: []
    };
  }

  componentDidMount = async () => {
    await this.props.getSavedJobs();
    this.setState({ savedJobs: this.props.savedJobs });
  }

  render() {
    const { isLoading, slideMenuActive, toggleMenu } = this.props;
    const { savedJobs } = this.state;

    return (
      <SlideMenu
        active={slideMenuActive}
        nav={nav}
        reactRouter={true}
        navClassName={"slide-menu"}
        closeMenu={() => toggleMenu()}
      >
        <div className="grid-container">
          <Header />
          <main className="content">
            <section className="cards-container">
              <h1 className="jobs-container__header">Saved Jobs</h1>
              {isLoading
                ?
                <div className="loader-container">
                  <Loader />
                </div>
                :
                <SavedJobs jobs={savedJobs} />
              }
            </section>
          </main>
          <Footer />
        </div>
      </ SlideMenu>
    );
  }
}

export const mapStateToProps = state => ({
  savedJobs: state.savedJobs,
  isLoading: state.isLoading,
  slideMenuActive: state.slideMenuActive
});

export const mapDispatchToProps = dispatch => ({
  getSavedJobs: () => dispatch(getSavedJobsFromFirebase()),
  toggleMenu: () => dispatch(toggleMenu())
});

export default connect(mapStateToProps, mapDispatchToProps)(JobsContainer);

JobsContainer.propTypes = {
  savedJobs: PropTypes.arrayOf(PropTypes.object),
  isLoading: PropTypes.bool,
  slideMenuActive: PropTypes.bool,
  getSavedJobs: PropTypes.func,
  toggleMenu: PropTypes.func
};