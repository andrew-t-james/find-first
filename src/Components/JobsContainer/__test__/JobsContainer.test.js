import React from 'react';
import { shallow } from 'enzyme';
import { JobsContainer } from '../JobsContainer';

describe('<JobsContainer />', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<JobsContainer />));

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
