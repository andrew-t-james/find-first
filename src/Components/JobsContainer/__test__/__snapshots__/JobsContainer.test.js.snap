// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobsContainer /> should match snapshot 1`] = `
<div
  className="grid-container"
>
  <Connect(Header) />
  <main
    className="content"
  >
    <section
      className="cards-container"
    >
      <h1>
        Jobs Container
      </h1>
      <CardContainer
        jobs={Array []}
      />
    </section>
  </main>
  <Connect(Footer) />
</div>
`;
